templates/templates.go

```diff
@@ -257,11 +259,17 @@ func (o *Output) Write() error {
 		return err
 	}
 
-	if o.Type == File {
+	switch o.Type {
+	case File:
 		return fileutil.WriteFile(path, o.Content, 0600)
+	case Snippet:
+		return fileutil.WriteSnippet(path, o.Content, 0600)
+	case Fragment:
+		lines := strings.Split(string(o.Content), "\n")
+		return fileutil.WriteFragment(path, o.Content, lines[0], lines[len(lines)-1], 0600)
```

Comment:
Instead of having the header/footer in the template can we just add it here:
```go
header := "# authority " + authorityName + " - autogenerated by step"
footer := "# end"
fileutil.WriteFragment(path, o.Content, header, footer, 0600)
```

I'm not sure if we're able to get the authority name here.